x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/FoldNull.java

```diff
@@ -9,7 +9,7 @@
 
 import org.elasticsearch.xpack.esql.core.expression.Expression;
 
-public class FoldNull extends LogicalPlanOptimizer.FoldNull {
+public class FoldNull extends org.elasticsearch.xpack.esql.optimizer.rules.FoldNull {
```

Comment:
Why does this class exist?  Really feels like we can delete it and just reference the superclass in all cases.

Reply:
Maybe it was for visibility issues in the past (the base class was not public in QL...?), but as long as it compiles, I guess we can get rid of this one now.